doc/api/events.md

```diff
@@ -155,9 +155,64 @@ myEmitter.emit('error', new Error('whoops!'));
 // Prints: whoops! there was an error
 ```
 
+## Capture rejections of Promises
+
+Using `async` functions with event handlers is problematic, because it would
+lead to an unhandled rejection in case of a thrown exception:
+
+```js
+const ee = new EventEmitter();
+ee.on('something', async (value) => {
+  throw new Error('kaboom');
+});
+```
+
+The `captureRejections` option in the `EventEmitter` constructor or the global
+setting change this behavior, installing a `.then(undefined, handler)`
+handler on the `Promise`. This handler routes the exception to the
+[`Symbol.for('nodejs.rejection')`][rejection] event if there is a listener,
+or to [`'error'`][error] handler if there is none.
+
+```js
+const ee1 = new EventEmitter({ captureRejections: true });
+ee1.on('something', async (value) => {
+  throw new Error('kaboom');
+});
+
+ee1.on('error', console.log);
+
+const ee2 = new EventEmitter({ captureRejections: true });
+ee2.on('something', async (value) => {
+  throw new Error('kaboom');
+});
+
+ee2.on(Symbol.for('nodejs.rejection'), console.log);
+```
+
+Setting `EventEmitter.captureRejections = true` will change the default for all
+new instances of `EventEmitter`.
+
+```js
+EventEmitter.captureRejections = true;
+const ee1 = new EventEmitter();
+ee1.on('something', async (value) => {
+  throw new Error('kaboom');
+});
+
+ee1.on('error', console.log);
+```
+
+The `Symbol.for('nodejs.rejection')` and `'error'` events that are
+generated by the `captureRejections` behavior **do not have a catch handler**
+to avoid infinite error loops.
```

Comment:
For these events, I *would* add a catch handler if `captureRejections` is enabled and would have those forward up to `process`. For `process`, if someone decided to do `process.on('error', async (err) => {})`, only then should it be handled as an unhandled rejection. 

Also, I would make it so that `process` is created with the `captureRejections` flag enabled by default